server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2584,6 +2585,53 @@ protected void runInContext() {
         }
     }
 
+    private void verifyVmLimits(UserVmVO vmInstance, Map<String,String> details) {
+        if (VirtualMachineManager.ResoureCountRunningVMsonly.value()) {
+            return;
+        }
+
+        long newCpu = NumberUtils.toLong(details.get(VmDetailConstants.CPU_NUMBER));
+        long newMemory = NumberUtils.toLong(details.get(VmDetailConstants.MEMORY));
```

Comment:
does _details_ contains both these params, or can be null?

Reply:
NumberUtil.toLong() -> verifies if the value is null and sets it to 0.

Reply:
OK, check if _decrementResourceCount()_ calls below are relavent when these values are null.